cpp/Osmosis/ObjectStore/DirectoryValidator.h

```diff
@@ -0,0 +1,49 @@
+#ifndef __OSMOSIS_OBJECT_STORE_DIRECTORY_VALIDATOR_H__
+#define __OSMOSIS_OBJECT_STORE_DIRECTORY_VALIDATOR_H__
+
+#include <boost/filesystem.hpp>
+#include <mutex>
+
+namespace Osmosis {
+namespace ObjectStore {
+
+class DirectoryValidator
```

Comment:
name too generic. how about "MakeSureDirectoryExists", or "DirectoryExistance", or even "MakeDirectory". DirectoryValidator sounds like there is something about it's structure that requires validating, but doesn't state why, so the code using this class can not be understood without reading this file...